src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2366,7 +2366,20 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
                                 salt = stack.PopBytes();
                             }
 
-                            long gasCost = GasCostOf.Create + (instruction == Instruction.CREATE2 ? GasCostOf.Sha3Word * EvmPooledMemory.Div32Ceiling(initCodeLength) : 0);
+                            //EIP-3860
+                            if (spec.IsEip3860Enabled && initCodeLength > spec.MaxInitCodeSize)
+                            {
+                                _returnDataBuffer = Array.Empty<byte>();
+                                stack.PushZero();
+                                break;
+                            }
+
+                            var initCodeWordLength = EvmPooledMemory.Div32Ceiling(initCodeLength);
```

Comment:
In theory this line changes current EVM behaviour, because EvmPooledMeory.Div32Ceiling can result with OutOfGasException

Reply:
Yes, that's corrected now.